source/grid/grid_tools.cc

```diff
@@ -4915,6 +4915,181 @@ namespace GridTools
 
 
 
+  template <int dim, int spacedim>
+  std::tuple< std::vector<typename Triangulation<dim, spacedim>::active_cell_iterator >, std::vector< std::vector< Point<dim> > >, std::vector<std::vector<unsigned int> > >
+  compute_point_locations(const Cache<dim,spacedim>                                         &cache,
+                          const std::vector<Point<spacedim> >                               &points,
+                          const double                                                      &distance_factor)
+  {
+    Assert(distance_factor >= 0.5,
+           ExcMessage("distance_factor value must be >= 0.5"));
+
+    std::tuple< std::vector<typename Triangulation<dim, spacedim>::active_cell_iterator >, std::vector< std::vector< Point<dim> > >, std::vector<std::vector<unsigned int> > >
+    cell_qpoint_map;
+
+    // How many points are here?
+    const unsigned int np = points.size();
+
+    // Now the easy case.
+    if (np==0) return cell_qpoint_map;
+
+    // Keep track of the points we
+    // found
+    std::vector<bool> point_flags(np, false);
+
+    // Classify the first point:
+    const std::pair<typename dealii::internal::ActiveCellIterator
+    <dim, dim, Triangulation<dim, spacedim>>::type, Point<dim> >
```

Comment:
Couldn't you use `Triangulation<dim,spacedim>::active_cell_iterator` as the first type here? That seems like a more readable name.